analysis.go

```diff
@@ -50,60 +49,51 @@ type DeploymentConfigs struct {
 	} `json:"items"`
 }
 
-type CheckTask func(string, io.Writer) (Result, error)
+type CheckTask func(string, string) (Result, error)
+
+func GetAnalysisTasks(tasks chan<- Task, basepath string, projects []string, results chan<- CheckResults) {
+	// Platform-wide analysis goes here
+
+	// project specific analysis in here
+	for _, p := range projects {
+		tasks <- CheckProjectTask(p, basepath, results)
+	}
+}
 
 // CheckTasks is a task factory for tasks that diagnose system conditions.
-func CheckTasks(project string, outFor, errOutFor projectResourceWriterCloserFactory) Task {
-	return checkTasks(func() []CheckTask {
+func CheckProjectTask(project, basepath string, results chan<- CheckResults) Task {
+	return checkProjectTask(func() []CheckTask {
 		return []CheckTask{CheckImagePullBackOff, CheckDeployConfigsReplicasNotZero}
-	}, project, outFor, errOutFor)
+	}, project, basepath, results)
 }
 
 // A getProjectCheckFactory generates tasks to diagnose system conditions.
 type getProjectCheckFactory func() []CheckTask
 
 type CheckResults struct {
+	Scope   string `json:"scope"`
```

Comment:
What is the `Scope` for? I'm missing some docs.
Why do we make it `"scope"` in JSON, but let `Results` use the default value (capitalized name)?